File: angga/transaction.pl
use LWP::UserAgent;
use JSON qw( decode_json );

my $url = "http://node_url";
my $private_key = $ENV{'PRIVATE_KEY'};
my $account = "angga";
my $actor = "angga";

sub perform_transaction {
    my ($name, $data) = @_;

    my $payload = {
        "actions" => [
            {
                "account" => $account,
                "name" => $name,
                "authorization" => [
                    {
                        "actor" => $actor,
                        "permission" => "active"
                    }
                ],
                "data" => $data
            }
        ]
    };

    my $ua = LWP::UserAgent->new;
    my $response = $ua->post( $url, Content => to_json($payload) );
    return decode_json( $response->content );
}

sub create_transaction {
    my ($id, $user, $data) = @_;
    my $data = { id => $id, user => $user, data => $data };
    return perform_transaction("create", $data);
}

sub read_transaction {
    my ($id) = @_;
    my $data = { id => $id };
    return perform_transaction("read", $data);
}

sub destroy_transaction {
    my ($id) = @_;
    my $data = { id => $id };
    return perform_transaction("destroy", $data);
}

sub push_transaction {
    my ($data_id, $user, $data) = @_;
    create_transaction($data_id, $user, $data);
    read_transaction($data_id);
    destroy_transaction($data_id);
}

push_transaction(1, $account, "This is RPC Push Transaction using Perl");

File: rafli/transaction.go
package main

import (
    "bytes"
    "encoding/json"
    "fmt"
    "net/http"
    "os"
)

var (
    url = "http://node_url"
    privateKey = os.Getenv("PRIVATE_KEY")
    account = "rafli"
    actor = "rafli"
)

type Payload struct {
    Actions []Action `json:"actions"`
}

type Action struct {
    Account       string          `json:"account"`
    Name          string          `json:"name"`
    Authorization []Authorization `json:"authorization"`
    Data          interface{}     `json:"data"`
}

type Authorization struct {
    Actor      string `json:"actor"`
    Permission string `json:"permission"`
}

func performTransaction(name string, data interface{}) (interface{}, error) {
    payload := Payload{
        Actions: []Action{
            {
                Account: account,
                Name:    name,
                Authorization: []Authorization{
                    {
                        Actor:      actor,
                        Permission: "active",
                    },
                },
                Data: data,
            },
        },
    }

    payloadJSON, err := json.Marshal(payload)
    if err != nil {
        return nil, err
    }

    resp, err := http.Post(url, "application/json", bytes.NewReader(payloadJSON))
    if err != nil {
        return nil, err
    }

    var responseJSON interface{}
    err = json.NewDecoder(resp.Body).Decode(&responseJSON)
    if err != nil {
        return nil, err
    }

    return responseJSON, nil
}

func createTransaction(id int, user string, data string) (interface{}, error) {
    dataMap := map[string]interface{}{
        "id":   id,
        "user": user,
        "data": data,
    }
    return performTransaction("create", dataMap)
}

func readTransaction(id int) (interface{}, error) {
    dataMap := map[string]interface{}{
        "id": id,
    }
    return performTransaction("read", dataMap)
}

func destroyTransaction(id int) (interface{}, error) {
    dataMap := map[string]interface{}{
        "id": id,
    }
    return performTransaction("destroy", dataMap)
}

func pushTransaction(dataID int, user string, data string) error {
    _, err := createTransaction(dataID, user, data)
    if err != nil {
        return err
    }
    _, err = readTransaction(dataID)
    if err != nil {
        return err
    }
    _, err = destroyTransaction(dataID)
    if err != nil {
        return err
    }
    return nil
}

func main() {
    err := pushTransaction(1, account, "This is RPC Push Transaction using Golang")
    if err != nil {
        fmt.Println(err)
    }
}
